### PR TITLE
[Documentation] Using non-OpenAI models

### PR DESCRIPTION
**Pull request by marklysze**
As there's a lot of interest in running AutoGen with non-OpenAI models and there's a drive to add it to the documentation (https://github.com/microsoft/autogen/issues/1994), here's a start to the documentation for using non-OpenAI models with cloud and local proxy servers.

There's a new menu item under `Docs` > `Topics` called `Using Non-OpenAI Models`.

Under this are the start of the documentation for using non-OpenAI models, with three pages:
1. Non-OpenAI Models (an introduction to using cloud and local proxy servers)
2. Cloud-based Proxy Servers (An example of a cloud proxy server, together.ai)
3. Locally Run Proxy Servers (An example of LiteLLM+Ollama)

I'll add vLLM to the locally run proxy servers and am also looking to add huggingface.co to the cloud-based (if I can get a working example).

It would be great for others to add examples for cloud-based and locally run proxy servers.

Adding custom model classes would also be a good additional page and examples.

Incorporating (https://github.com/microsoft/autogen/pull/2044 - LM Studio example) by @ekzhu is necessary, too, and that has a matching folder structure.

## Related issue number

[1994 - [Documentation] Topic Category for AutoGen + Non-OpenAI Models](https://github.com/microsoft/autogen/issues/1994) is the driver for this PR.

[2044 - Add LM Studio Example in Topics](https://github.com/microsoft/autogen/issues/1994) to be incorporated / merged.

## Checks

- [X] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

## UPDATE
The current documentation structure is now:
Docs > Topics > Using Non-OpenAI Models

Under this is an introduction page and then individual pages for cloud-based and local proxy servers.

### CODE REVIEW

File: website/docs/topics/non-openai-models/cloud-based-proxy-servers.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+When cloud-based proxy servers provide an OpenAI-compatible API, using them in AutoGen
+is straightforward. With [LLM Configuration](/docs/topics/llm_configuration) done in the same way
+as when using OpenAI's models, the primary difference is typically the authentication which is
+usually handled through an API key.
+
+An example of a cloud-based proxy server provider that has an OpenAI-compatible API is provided
+below.
+
+## Together AI
+![together.ai](images/togetherai250.png)
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/cloud-based-proxy-servers.md:11]: We don't need to add their logo here for now as their logos can change overtime.

File: website/docs/topics/non-openai-models/cloud-based-proxy-servers.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+This example, using [together.ai](https://www.together.ai/), is a group chat between a Python developer
+and a code reviewer, who are given a coding task.
+
+Start by [installing AutoGen](/docs/installation/) and getting your [together.ai API key](https://api.together.xyz/settings/profile).
+
+Put your together.ai API key in an environment variable, TOGETHER_API_KEY.
+
+Linux / Mac OSX:
+
+```python
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/cloud-based-proxy-servers.md:22]:
```suggestion
```bash
```

File: website/docs/topics/non-openai-models/cloud-based-proxy-servers.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+$ export TOGETHER_API_KEY=YourTogetherAIKeyHere
+```
+
+Windows (command prompt):
+
+```python
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/cloud-based-proxy-servers.md:28]:
```suggestion
```powershell
```

File: website/docs/topics/non-openai-models/cloud-based-proxy-servers.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+and a code reviewer, who are given a coding task.
+
+Start by [installing AutoGen](/docs/installation/) and getting your [together.ai API key](https://api.together.xyz/settings/profile).
+
+Put your together.ai API key in an environment variable, TOGETHER_API_KEY.
+
+Linux / Mac OSX:
+
+```python
+$ export TOGETHER_API_KEY=YourTogetherAIKeyHere
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/cloud-based-proxy-servers.md:23]:
```suggestion
export TOGETHER_API_KEY=YourTogetherAIKeyHere
```

File: website/docs/topics/non-openai-models/local-proxy-servers.md [hunk 5]
```diff
@@ -0,0 +1,492 @@
+# Locally Run Proxy Servers
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/local-proxy-servers.md:1]: This pape contains two example of local run proxy server and it is already a bit long. 
We can split this page into "LiteLLM and Ollama" and "vLLM", and we can move the first two sentences in this page to the `about-using-nonopenai-models.md` and list the splitted pages there.

File: website/docs/topics/non-openai-models/about-using-nonopenai-models.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+![Cloud or Local Proxy Servers](images/cloudlocalproxy.png)
+
+### Cloud-based proxy servers
+By using cloud-based proxy servers, you are able to use models without requiring the hardware
+and software to run them.
+
+These providers can host open source/weight models, like [Hugging Face](https://huggingface.co/),
+or their own closed models.
+
+See [Cloud-based Proxy Servers](cloud-based-proxy-servers) for a guide on how to
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/about-using-nonopenai-models.md:27]: We can list the pages on cloud based servers here. People can add more in the future.

File: website/docs/topics/non-openai-models/about-using-nonopenai-models.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+use them with AutoGen.
+
+### Locally run proxy servers
+An increasing number of LLM proxy servers are available for use locally. These can be 
+open-source (e.g., LiteLLM, Ollama, vLLM) or closed-source (e.g., LM Studio), and are
+typically used for running the full-stack within your environment.
+
+See [Local Proxy Servers](local-proxy-servers) for a guide on how to use them with
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/about-using-nonopenai-models.md:35]: Same as above, use a list of local server pages here.

File: website/docs/topics/non-openai-models/cloud-based-proxy-servers.md [hunk 8]
```diff
@@ -0,0 +1,187 @@
+# Cloud-based Proxy Servers
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/cloud-based-proxy-servers.md:1]: Same comment as in the local proxy server: we can rename this page "Together AI", and move the first two paragraphs here to the `about-using-non-openai-models.md`.

File: website/docs/topics/non-openai-models/local-proxy-servers.md [hunk 9]
```diff
@@ -0,0 +1,492 @@
...
+Not all open source/weight models are suitable for function calling and AutoGen continues to be
+developed to provide wider support for open source models.
+
+The [#alt-models](https://discord.com/channels/1153072414184452236/1201369716057440287) channel
+on AutoGen's Discord is an active community discussing the use of open source/weight models
+with AutoGen.
+:::
+````
+
+## vLLM
```
Comment (ekzhu) [inline website/docs/topics/non-openai-models/local-proxy-servers.md:330]: We can split the vLLM part to a separate page named "vLLM".

**Review by ekzhu — CHANGES_REQUESTED** [review]
Thanks for the PR! My comments are mostly about organizing the content into smaller parts for easier navigation and better indexibiity.

File: website/docs/topics/non-openai-models/about-using-nonopenai-models.md (hunk 6 above)
Comment (marklysze) [reply website/docs/topics/non-openai-models/about-using-nonopenai-models.md:27]: Good suggestion, I've put the cloud and local proxy server examples into their own pages and referenced it from this page.

**Comment by ekzhu** [comment]
Thanks! To run pre-commit formatting check locally, you can do:

```
pre-commit run --all-files
```

File: website/docs/topics/non-openai-models/local-lm-studio.ipynb
Comment (ekzhu) [inline website/docs/topics/non-openai-models/local-lm-studio.ipynb:1]: Can you add a redirec from `lm-studio` to `local-lm-studio` here:

https://github.com/microsoft/autogen/blob/4af04407302c6d2e8ae24161772af2f9ea87ccc9/website/docusaurus.config.js#L191

**Review by ekzhu — APPROVED** [review]

**Pull request merged**